src/Parser.js

```diff
@@ -13,7 +13,7 @@ export function compile (value) {
 /**
  * @param {string} value
  * @param {object} root
- * @param {object|null} parent
+ * @param {?object} parent
```

Comment:
I've been using the {type?} suffix placement style.

Reply:
K, adjusted it to the proposed syntax